src/orb.yml.hbs

```diff
@@ -848,42 +848,51 @@ commands:
           Only in use when verify-revision-is-deployed is set to true.
         type: boolean
         default: true
+      run-update-task-definition:
+        description: |
+          Whether to run the update-task-definition again in the update-service or
+          not. Defaults to false and will only run when parameter is set to true.
+        type: boolean
+        default: false
     steps:
-      - update-task-definition:
-          family: << parameters.family >>
-          container-image-name-updates: << parameters.container-image-name-updates >>
-          container-env-var-updates: << parameters.container-env-var-updates >>
-      - run:
-          name: Update service with registered task definition
-          command: |
-            DEPLOYMENT_CONTROLLER="$(echo << parameters.deployment-controller >>)"
+      - when:
+          condition: <<parameters.run-update-task-definition>>
+          steps:
+            - update-task-definition:
+                family: << parameters.family >>
+                container-image-name-updates: << parameters.container-image-name-updates >>
+                container-env-var-updates: << parameters.container-env-var-updates >>
+            - run:
```

Comment:
I think the indentation here should be changed, to prevent also skipping the service update when a new task definition is not created.

Also, this step relies on the `CCI_ORB_AWS_ECS_REGISTERED_TASK_DFN` environment variable being defined, which is defined by `update-task-definition`. Hence my suggestion is to insert a new step in-between, that runs only if `update-task-definition` is skipped, that will retrieve the last task definition and use it to define `CCI_ORB_AWS_ECS_REGISTERED_TASK_DFN`.

For example:

```
TASK_DEFINITION_ARN=$(aws ecs describe-task-definition \
  --task-definition << parameters.family >> \
  --output text \
  --query 'taskDefinition.taskDefinitionArn')
echo "export CCI_ORB_AWS_ECS_REGISTERED_TASK_DFN='${TASK_DEFINITION_ARN}'" >> $BASH_ENV
```

Reply:
@Xheno I think there is still an issue with the service update being skipped here. I'm going to push some fixes for this 🙂